### ISSUE TITLE
Duplication character in char field

### ISSUE THREAD

**Issue by claudsan** [hgourvest/node-firebird#162]
When you bring the rows that have fields of type char, it is doubling the amount of characters in the result.

**Example:** 

The field 'TEST1 char (1)', with the value 'A' 
 
returns in json the value ```'A  '```, with 2 spaces after the character A.

And field 'TEST2 char(3)', with the value 'B'
returns in json the value  ```'B         ' ```, with 9 spaces after the character B.

 ```
Firebird.attach(options, (err, db) => {
            if (err)
                throw err;

            db.query("SELECT * FROM TABLE_CHAR",function(err, result) {
                console.log(result);
                db.detach();
            });
            console.log("fim");
        });

//Table
//CREATE TABLE TEST_CHAR (
//    TEST1 CHAR(1),
//    TEST2 CHAR(3))
```
in **lib\index.js** line 814 in **SQLVarText.prototype.decode**
```
SQLVarText {
  type: 452,
  nullable: false,
  subType: 4,
  scale: 0,
  length: 4,
  field: 'TEST1',
  relation: 'TEST_CHAR',
  alias: 'TEST1' }
```
the attribute length is wrong only in char types.  

Solution for me, in line 814 

> ret = data.readText(this.length, DEFAULT_ENCODING).trim();

**but it did not look elegant**

Server version: **LI-V6.3.2.32703 Firebird 3.0**
OS: **Ubuntu 16.04.3**

**Comment by joweste** [comment]
Hi, I have the same problem. It becomes very hard make and API with this problem.
This is not the normal behaviour with others DB as mysql and postgree

**Comment by mariuz** [comment]
Verified on the current driver: `CHAR(1)` with 'A' returns exactly `'A'` (and `CHAR(5)` returns the SQL-standard space padding to its declared length, e.g. `'AB   '` — use `TRIM()` or `VARCHAR` if padding is unwanted). The multi-byte-charset over-read that doubled characters in that era is fixed; the decode now truncates to the column's character length. Closing — reopen with charset + DDL if you still see doubling on 2.x.
